quickget

```diff
@@ -3326,6 +3286,72 @@ function create_vm() {
     make_vm_config "${ISO}"
 }
 
+function create_config() {
+    local VM_PATH="${1}"
+    local INPUT="${2}"
+    local FIXED_ISO=""
+
+    OS="custom"
+    if ! mkdir "${VM_PATH}" 2>/dev/null; then
+        echo "ERROR! Could not create directory: ${VM_PATH}. Please verify that it does not already exist"
+        exit 1
+    fi
+    if [[ "${INPUT}" == "http://"* ]] || [[ "${INPUT}" == "https://"* ]]; then
+        INPUT="$(web_redirect "${INPUT}")"
+        if [[ "${INPUT}" == *".iso" ]] || [[ "${INPUT}" == *".img" ]]; then
+            web_get "${INPUT}" "${VM_PATH}"
+            INPUT="${VM_PATH}/${INPUT##*/}"
+        else
+            echo "ERROR! Only ISO,IMG and QCOW2 file types are supported for --create-config"
+            exit 1
+        fi
+    fi
+    if [[ "${INPUT}" == *".iso" ]]; then
+        echo "Moving image to VM dir" && mv "${INPUT}" "${VM_PATH}"
+        CUSTOM_IMAGE_TYPE="iso"
+    elif [[ "${INPUT}" == *".img" ]]; then
+        echo "Moving image to VM dir" && mv "${INPUT}" "${VM_PATH}"
+        CUSTOM_IMAGE_TYPE="img"
+    elif [[ "${INPUT}" == *".qcow2" ]]; then
+        echo "Moving image to VM dir" && mv "${INPUT}" "${VM_PATH}/disk.qcow2"
+        CUSTOM_IMAGE_TYPE="qcow2"
+    else
+        echo "ERROR! Only ISO,IMG and QCOW2 file types are supported for --create-config"
+        exit 1
+    fi
+    echo "Creating custom VM config for ${INPUT##*/}."
+    case "${INPUT,,}" in
+        *windows-server*) CUSTOM_OS="windows-server";;
+        *windows*)
+            CUSTOM_OS="windows"
+            while [ $# -gt 2 ]; do
+                case "${3}" in
+                    -unattended|--unattended)
+                        echo "Downloading VirtIO drivers..."
+                        web_get "https://fedorapeople.org/groups/virt/virtio-win/direct-downloads/stable-virtio/virtio-win.iso" "${VM_PATH}"
+                        FIXED_ISO=${VM_PATH}/virtio-win.iso
+
+                        rm -f "${VM_PATH}/unattended.iso"
+                        unattended_windows "${VM_PATH}"
+                        shift
+                esac
+            done
+            ;;
```

Comment:
First off, I have no idea why we're using a loop for this. There are no parameters that could possibly be passed here other than your proposed --unattended. In addition, it's an infinite loop if there's a third argument that's not --unattended, since shift is only executed if the value is --unattended.

I also disagree that we should require an extra parameter for this, unattended installation should be the default and automatically handled for the user if possible. Therefore, I propose using pattern matching to determine if the specified ISO has a name matching Microsoft's formatting for consumer Windows ISOs, with optional flags to override it

Please validate the code and formatting before including it, of course. It may not be completely correct. I've written it in the GitHub comment interface rather than within my editor.

```suggestion
            # After checking flags, determine whether ISO is named as a Windows 10 or 11 ISO using pattern matching
            # Older windows 10 ISOs use the year followed by the month rather than the year & half). Match any text for language.
            if [ "${3}" != "--disable-unattended" ] && ( [ "${3}" == "--unattended" ]
                || grep -E -q 'Win(10|11)_([0-9]{2}H(1|2)|[0-9]{4})_[^.]*?(x64|x32)(v[0-9])?.iso' <<< "${INPUT}" ); then
                echo "Downloading VirtIO drivers..."
                web_get "https://fedorapeople.org/groups/virt/virtio-win/direct-downloads/stable-virtio/virtio-win.iso" "${VM_PATH}"
                FIXED_ISO=${VM_PATH}/virtio-win.iso

                rm -f "${VM_PATH}/unattended.iso"
                unattended_windows "${VM_PATH}"
            fi
```

Reply:
> First off, I have no idea why we're using a loop for this.
The reason i used a loop is for extensibility. It will be easier to add additional flags for specific type of installations if required in the future.

> I also disagree that we should require an extra parameter for this, unattended installation should be the default and automatically handled for the user if possible. 

Thats absolutely correct. My proposed solution requires users to rename their iso to "windows.iso". Having pattern matching would simply be better.

Reply:
> My proposed solution requires users to rename their iso to "windows.iso". Having pattern matching would simply be better.

No, it doesn't. I'm not sure what you're talking about. Your solution requires users to pass "--unattended" after to set up unattended setup. I've already fixed the issue where quickget wouldn't detect Windows ISOs using standard naming in 5d64018, which I've committed directly to your branch due to it wasting a bunch of space in conversations for issues that don't have to do with your proposed changes.

See my suggested changes, If the flags --disable-unattended or --unattended aren't present, with my changes, quickget would check whether the ISO follows Microsoft's naming scheme for windows 10 or 11 ISOs using a regex. This would remove the need to pass --unattended as long as you don't rename the ISO. 

To explain the syntax used.
Win(10|11) - pretty self explanatory, matches Win and then 10 or 11
_([0-9]{2}H(1|2)|[0-9]{4}) - matches either 2 digits (year), H, and then 1 or 2 (halves of year) (e.g. 23H2), or 4 digits (year + month) for older Windows 10 ISOs (e.g. 1607)
_[^.]*? - matches an underscore and then any character other than a period or specific other characters such as newlines. This allows us to skip languages, which can be formatted in many ways, with multiple underscores.
(x64|x32) - self explanatory.  Matches either x64 or x32, Microsoft's naming for the x86_64 and i686 architectures
(v[0-9])? - optionally matches v followed by one digit, for versions of some Windows ISOs

Reply:
> No, it doesn't. I'm not sure what you're talking about.

I apologize to have quoted the wrong block of text. Let me correct that:

> Therefore, I propose using pattern matching to determine if the specified ISO has a name matching Microsoft's formatting for consumer Windows ISOs

Thats absolutely correct. My proposed solution requires users to rename their iso to "windows.iso" (which is not optimal). Having pattern matching would simply be better (which is your proposed solution).

> the point of quickemu is that you don't have to read a manual to set up QEMU

I agree with that. Lets use your solution then.